pkg/tsdb/azuremonitor/aztokenprovider/token_provider.go

```diff
@@ -112,7 +112,7 @@ func resolveAuthorityForCloud(cloudName string) string {
 	case setting.AzureUSGovernment:
 		return azidentity.AzureGovernment
 	case setting.AzureGermany:
-		return azidentity.AzureGermany
+		return ""
```

Comment:
Seems to no longer be defined within the `azidentity` package. 

Should we fallback to `azidentity.AzurePublicCloud`? 🤔

Reply:
[that cloud seems to not be available since Oct 21](https://docs.microsoft.com/en-us/azure/germany/germany-welcome) so I think it's fine to remove this case.

Reply:
Yes, please remove the case for Germany

Reply:
Anything needed to handle the removal of this case for backward compatibility reasons or is "empty" string the fallback mechanism?

Reply:
The current logic is not correct, since "" means the default authority e.g. the same thing as `azidentity.AzurePublicCloud`.
The proper logic would be to fail with `The cloud not supported` error.

```go
default:
	err := fmt.Errorf("the cloud '%s' not supported", cloudName)
	return "", err
```

But you don't need to fix it in this PR. It's fine to fallback to default value.